### PR TITLE
ta/README: update link to talloc documentation

### PR DESCRIPTION
**Pull request by maxdunbar**
Cleaned up a sentence and updated the link to the talloc API documentation.

### CODE REVIEW

File: ta/README [hunk 1]
```diff
@@ -31,7 +31,7 @@ will be replaced with TA calls, and the talloc wrapper will be removed.
 
 Documentation for the talloc API is here:
 
-    http://talloc.samba.org/talloc/doc/html/modules.html
+    https://talloc.samba.org/talloc/doc/html/group__talloc.html
```
[the remark this answers is not in the document]
Comment (maxdunbar) [reply ta/README:34]: The subpage shows everything the modules.html has and more. group__talloc.html seems more like the actual main page of the documentation.

File: ta/README [hunk 2]
```diff
@@ -2,7 +2,7 @@ TA ("Tree Allocator") is a wrapper around malloc() and related functions,
 adding features like automatically freeing sub-trees of memory allocations if
 a parent allocation is freed.
 
-Generally, the idea is that every TA allocation can have a parent (indicated
+Generally, every TA allocation can have a parent (indicated
```
[the remark this answers is not in the document]
Comment (maxdunbar) [reply ta/README:5]: Yeah, its up to you, I just felt like the sentence felt a bit wordy for documentation.

File: ta/README (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dudemanguy) [reply ta/README:34]: Oh you're right, my bad. It's an overview. I mistakenly thought it was a specific subsection.

File: ta/README (hunk 2 above)
[the remark this answers is not in the document]
Comment (Dudemanguy) [reply ta/README:5]: I think I prefer the original personally. We can stick to updating the url.
